### PR TITLE
Disable control flow e2e ADF

### PR DESCRIPTION
**Pull request by eromano**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [x] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
This PR disable the controll flow in ADF

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by adinapitul** [comment]
1961 files changed?
shouldn't out-tsc folder be git ignored?

**Comment by eromano** [comment]
> 1961 files changed?
> shouldn't out-tsc folder be git ignored?

Yep I will check

File: e2e/process-services-cloud/task-form-cloud-component.e2e.ts [hunk 1]
```diff
@@ -50,21 +50,31 @@ describe('Task form cloud component', () => {
...
         await apiService.login(browser.params.identityAdmin.email, browser.params.identityAdmin.password);
 
         identityService = new IdentityService(apiService);
+        groupIdentityService = new GroupIdentityService(apiService);
+
         testUser = await identityService.createIdentityUserWithRole(apiService, [identityService.ROLES.APS_USER]);
+
+        const groupInfo = await groupIdentityService.getGroupInfoByGroupName('hr');
+        await identityService.addUserToGroup(testUser.idIdentityService, groupInfo.id);
+        await apiService.login(testUser.email, testUser.password);
```
Comment (adinapitul) [inline e2e/process-services-cloud/task-form-cloud-component.e2e.ts]: duplicate call

Comment (eromano) [reply e2e/process-services-cloud/task-form-cloud-component.e2e.ts]: done

File: e2e/content-services/directives/create-library-directive.e2e.ts [hunk 2]
```diff
@@ -57,222 +57,221 @@ describe('Create library directive', function () {
...
 
-    it('[C290176] Should not accept special characters for Library Id', () => {
+    it('[C290176] Should not accept special characters for Library Id', async () => {
         const name = 'My Library';
         const libraryId: string[] = ['My New Library', 'My+New+Library123!', '<>'];
 
-        createLibraryDialog.typeLibraryName(name);
+        await createLibraryDialog.typeLibraryName(name);
 
         for (let _i = 0; _i < 3; _i++) {
```
Comment (adinapitul) [inline e2e/content-services/directives/create-library-directive.e2e.ts]: underscore can be removed

File: lib/testing/src/lib/process-services/pages/start-process-dialog.page.ts [hunk 3]
```diff
@@ -18,26 +18,27 @@
 import { BrowserVisibility } from '../../core/utils/browser-visibility';
 import { by, element } from 'protractor';
 import { BrowserActions } from '../../core/utils/browser-actions';
+import { ElementFinder } from 'protractor/built/element';
```
Comment (adinapitul) [inline lib/testing/src/lib/process-services/pages/start-process-dialog.page.ts]: correct import would be from 'protractor' , though I don't think this exact one causes harm

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts [hunk 4]
```diff
@@ -39,120 +39,119 @@ export class TaskListCloudComponentPage {
 
     dataTable = new DataTableComponentPage(this.taskList);
 
-    getDataTable() {
+    getDataTable(): DataTableComponentPage {
         return this.dataTable;
     }
 
-    clickCheckbox(taskName) {
+    clickCheckbox(taskName): Promise<void> {
```
Comment (adinapitul) [inline lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: needs async ?

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts [hunk 5]
```diff
@@ -39,120 +39,119 @@ export class TaskListCloudComponentPage {
...
         return this.dataTable.clickCheckbox(column.name, taskName);
     }
 
-    checkRowIsNotChecked(taskName) {
+    checkRowIsNotChecked(taskName): Promise<void> {
```
Comment (adinapitul) [inline lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: async

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts [hunk 6]
```diff
@@ -39,120 +39,119 @@ export class TaskListCloudComponentPage {
...
         return this.dataTable.checkRowIsNotChecked(column.name, taskName);
     }
 
-    checkRowIsChecked(taskName) {
+    checkRowIsChecked(taskName): Promise<void> {
```
Comment (adinapitul) [inline lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: async

File: lib/testing/src/lib/core/pages/form/widgets/displayTextWidget.ts [hunk 7]
```diff
@@ -16,23 +16,23 @@
...
 
-    formFields = new FormFields();
-    labelLocator = by.css('div[class*="adf-display-text-widget"]');
-    inputLocator = by.css('input');
+    formFields: FormFields = new FormFields();
+    labelLocator: Locator = by.css('div[class*="adf-display-text-widget"]');
+    inputLocator: Locator = by.css('input');
 
-    getFieldLabel(fieldId) {
+    getFieldLabel(fieldId): Promise<string> {
```
Comment (adinapitul) [inline lib/testing/src/lib/core/pages/form/widgets/displayTextWidget.ts]: async for code consistency with all other pages

File: lib/testing/src/lib/core/pages/form/widgets/dateTimeWidget.ts [hunk 8]
```diff
@@ -16,81 +16,83 @@
...
     }
 
-    setDateTimeInput(fieldId, value) {
-        return this.formFields.setValueInInputById(fieldId, value);
+    async setDateTimeInput(fieldId, value): Promise<void> {
+        await this.formFields.setValueInInputById(fieldId, value);
     }
 
-    clearDateTimeInput(fieldId) {
+    async learDateTimeInput(fieldId): Promise<void> {
```
Comment (adinapitul) [inline lib/testing/src/lib/core/pages/form/widgets/dateTimeWidget.ts]: missing "c" from "clear" => duplicate correct method is at the end of the file

File: lib/testing/src/lib/process-services/pages/form-fields.page.ts [hunk 9]
```diff
@@ -16,144 +16,138 @@
...
-        BrowserVisibility.waitUntilElementIsVisible(input);
-        input.clear().sendKeys(value);
-        return this;
+        await BrowserVisibility.waitUntilElementIsVisible(input);
+        await BrowserActions.clearSendKeys(input, value);
     }
 
-    isCompleteFormButtonDisabled() {
-        BrowserVisibility.waitUntilElementIsVisible(this.completeButton);
+    async isCompleteFormButtonDisabled(): Promise<string> {
```
Comment (adinapitul) [inline lib/testing/src/lib/process-services/pages/form-fields.page.ts]: according to the method name, I would expect a return type of Promise<boolean>

File: lib/testing/src/lib/core/pages/form/formFields.ts [hunk 10]
```diff
@@ -15,149 +15,140 @@
...
+    async setValueInInputById(fieldId, value): Promise<void> {
         const input = element(by.id(fieldId));
-        BrowserActions.clearSendKeys(input, value);
-        return this;
+        await BrowserActions.clearSendKeys(input, value);
     }
 
-    isCompleteFormButtonDisabled() {
-        BrowserVisibility.waitUntilElementIsVisible(this.completeButton);
+    async isCompleteFormButtonDisabled(): Promise<string> {
```
Comment (adinapitul) [inline lib/testing/src/lib/core/pages/form/formFields.ts]: according to the name, return type should be Promise<boolean>

File: e2e/pages/adf/viewerPage.ts [hunk 11]
```diff
@@ -15,666 +15,619 @@
...
 
-    checkFullScreenButtonIsDisplayed() {
-        BrowserVisibility.waitUntilElementIsVisible(this.fullScreenButton);
+    async checkMediaPlayerContainerIsDisplayed(): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.mediaContainer);
     }
 
-    checkFullScreenButtonIsNotDisplayed() {
-        BrowserVisibility.waitUntilElementIsNotVisible(this.fullScreenButton);
+    async checkFileContent(pageNumber, text): Promise<void> {
```
Comment (adinapitul) [inline e2e/pages/adf/viewerPage.ts]: parameters types ?

File: e2e/pages/adf/viewerPage.ts [hunk 12]
```diff
@@ -15,666 +15,619 @@
...
-    enterCustomName(text) {
-        const textField = element(by.css('input[data-automation-id="adf-text-custom-name"]'));
-        BrowserVisibility.waitUntilElementIsVisible(textField);
-        textField.sendKeys('');
-        textField.clear();
-        textField.sendKeys(text);
-        return this;
+    async enterCustomName(text): Promise<void> {
+        const textField: ElementFinder = element(by.css('input[data-automation-id="adf-text-custom-name"]'));
+        await BrowserVisibility.waitUntilElementIsVisible(textField);
```
Comment (adinapitul) [inline e2e/pages/adf/viewerPage.ts]: not needed. wait for visibility already done in clearSendKeys

File: e2e/pages/adf/process-services/processServiceTabBarPage.ts [hunk 13]
```diff
@@ -16,33 +16,34 @@
...
+    processButton: ElementFinder = element.all(by.cssContainingText('div[class*="mat-tab-label"] .mat-tab-labels div', 'Process')).first();
+    reportsButton: ElementFinder = element.all(by.cssContainingText('div[class*="mat-tab-label"] .mat-tab-labels div', 'Reports')).first();
+    settingsButton: ElementFinder = element.all(by.cssContainingText('div[class*="mat-tab-label"] .mat-tab-labels div', 'Settings')).first();
+    reportsButtonSelected: ElementFinder = element.all(by.cssContainingText('div[class*="mat-tab-label"] .mat-tab-labels div[aria-selected="true"]', 'Reports')).first();
 
-    clickTasksButton() {
-        BrowserActions.click(this.tasksButton);
-        browser.sleep(700);
+    async clickTasksButton(): Promise<void> {
+        await BrowserActions.closeMenuAndDialogs();
```
Comment (adinapitul) [inline e2e/pages/adf/process-services/processServiceTabBarPage.ts]: closing other possible left over dialogs should be the responsibility of the test
this method should do what the name says, just click on the Tasks button

File: e2e/pages/adf/contentServicesPage.ts [hunk 14]
```diff
@@ -35,166 +35,158 @@ export class ContentServicesPage {
...
+    async checkContextActionIsVisible(actionName) {
+        const actionButton: ElementFinder = element(by.css(`button[data-automation-id="context-${actionName}"`));
+        await BrowserVisibility.waitUntilElementIsVisible(actionButton);
     }
 
-    checkContentActionIsEnabled(actionName) {
-        const actionButton = element(by.css(`button[data-automation-id="context-${actionName}"`));
-        BrowserVisibility.waitUntilElementIsVisible(actionButton);
-        return actionButton.isEnabled();
+    async checkContextActionIsEnabled(actionName): Promise<boolean> {
```
Comment (adinapitul) [inline e2e/pages/adf/contentServicesPage.ts]: name is misleading, suggest: isContextActionEnabled

File: e2e/pages/adf/contentServicesPage.ts [hunk 15]
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
...
+    async sortAndCheckListIsOrderedByCreated(sortOrder: string): Promise<any> {
+        await this.sortByCreated(sortOrder);
+        return await this.checkListIsSortedByCreatedColumn(sortOrder);
     }
 
-    doubleClickRow(nodeName) {
-        this.contentList.doubleClickRow(nodeName);
-        return this;
+    async doubleClickRow(nodeName): Promise<void> {
+        await BrowserActions.closeMenuAndDialogs();
```
Comment (adinapitul) [inline e2e/pages/adf/contentServicesPage.ts]: this should be the responsibility of the test
this method should have a single responsibility: double clicking on an element.

File: e2e/pages/adf/contentServicesPage.ts [hunk 16]
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
...
-    clickOnEditFolder() {
-        BrowserActions.click(this.editFolderButton);
-        return this;
+    async clickOnEditFolder(): Promise<void> {
+        await BrowserActions.click(this.editFolderButton);
     }
 
-    checkEditFolderButtonIsEnabled() {
-        return this.editFolderButton.isEnabled();
+    async checkEditFolderButtonIsEnabled(): Promise<boolean> {
```
Comment (adinapitul) [inline e2e/pages/adf/contentServicesPage.ts]: name doesn't match return type, suggest: isEditFolderButtonEnabled

File: e2e/pages/adf/contentServicesPage.ts [hunk 17]
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
...
+        return await this.uploadFileButton.isEnabled();
     }
 
-    enableCustomPermissionMessage() {
-        const customPermissionMessage = element(by.cssContainingText('.mat-slide-toggle-content', 'Enable custom permission message'));
-        BrowserActions.click(customPermissionMessage);
-        return this;
+    async getErrorMessage(): Promise<string> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.errorSnackBar);
+        return await this.errorSnackBar.getText();
```
Comment (adinapitul) [inline e2e/pages/adf/contentServicesPage.ts]: BrowserActions could be used here

File: e2e/pages/adf/contentServicesPage.ts [hunk 18]
```diff
@@ -229,459 +221,396 @@ export class ContentServicesPage {
...
     }
 
-    selectGridSortingFromDropdown(sortingChosen) {
-        BrowserActions.closeMenuAndDialogs();
-        const dropdownSorting = this.getGridViewSortingDropdown();
-        BrowserActions.click(dropdownSorting);
-        const optionToClick = element(by.css(`mat-option[data-automation-id="grid-view-sorting-${sortingChosen}"]`));
-        BrowserActions.click(optionToClick);
+    async selectGridSortingFromDropdown(sortingChosen): Promise<void> {
+        // await BrowserActions.closeMenuAndDialogs();
```
Comment (adinapitul) [inline e2e/pages/adf/contentServicesPage.ts]: remove comment if not needed

File: e2e/pages/adf/content-services/breadcrumb/breadCrumbDropdownPage.ts [hunk 19]
```diff
@@ -15,32 +15,32 @@
...
-    clickParentFolder() {
-        BrowserActions.click(this.parentFolder);
+    async clickParentFolder(): Promise<void> {
+        await BrowserActions.click(this.parentFolder);
     }
 
-    checkBreadCrumbDropdownIsDisplayed() {
-        BrowserVisibility.waitUntilElementIsVisible(this.breadCrumbDropdown);
+    async checkBreadCrumbDropdownIsDisplayed(): Promise<void> {
+        await BrowserVisibility. waitUntilElementIsVisible(this.breadCrumbDropdown);
```
Comment (adinapitul) [inline e2e/pages/adf/content-services/breadcrumb/breadCrumbDropdownPage.ts]: extra space

File: e2e/pages/adf/configEditorPage.ts [hunk 20]
```diff
@@ -15,35 +15,33 @@
...
+    textField: ElementFinder = element(by.css('#adf-form-config-editor div.overflow-guard > textarea'));
 
-    enterConfiguration(text) {
-
-        BrowserVisibility.waitUntilElementIsVisible(this.textField);
-        this.textField.sendKeys(text);
-        return this;
+    async enterConfiguration(text): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.textField);
+        await this.textField.sendKeys(text);
```
Comment (adinapitul) [inline e2e/pages/adf/configEditorPage.ts]: BrowserActions could be used

File: e2e/pages/adf/commentsPage.ts [hunk 21]
```diff
@@ -15,59 +15,59 @@
...
 
-    getTotalNumberOfComments() {
+    async getTotalNumberOfComments(): Promise<string> {
         return BrowserActions.getText(this.numberOfComments);
     }
 
-    checkUserIconIsDisplayed(position) {
-        BrowserVisibility.waitUntilElementIsVisible(this.commentUserIcon.first());
+    async checkUserIconIsDisplayed(position): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.commentUserIcon.first());
```
Comment (adinapitul) [inline e2e/pages/adf/commentsPage.ts]: the return on this method doesn't make sense.

File: e2e/pages/adf/commentsPage.ts [hunk 22]
```diff
@@ -15,59 +15,59 @@
...
+        await BrowserVisibility.waitUntilElementIsNotVisible(this.commentInput);
     }
 
-    addComment(comment) {
-        BrowserVisibility.waitUntilElementIsVisible(this.commentInput);
-        this.commentInput.sendKeys(comment);
-        BrowserActions.click(this.addCommentButton);
+    async addComment(comment): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.commentInput);
+        await this.commentInput.sendKeys(comment);
```
Comment (adinapitul) [inline e2e/pages/adf/commentsPage.ts]: BrowserActions could be used

File: e2e/pages/adf/cardViewComponentPage.ts [hunk 23]
```diff
@@ -15,182 +15,166 @@
...
         return BrowserActions.getText(errorElement);
     }
 
-    setName(name) {
-        BrowserVisibility.waitUntilElementIsVisible(this.nameInputField);
-        this.nameInputField.sendKeys(name);
-        return this;
+    async setName(name): Promise<void> {
+        await BrowserVisibility.waitUntilElementIsVisible(this.nameInputField);
+        await this.nameInputField.sendKeys(name);
```
Comment (adinapitul) [inline e2e/pages/adf/cardViewComponentPage.ts]: BrowserActions could be used

File: e2e/content-services/document-list/document-list-gallery.e2e.ts [hunk 24]
```diff
@@ -37,7 +37,7 @@ describe('Document List Component', () => {
     const navBar = new NavigationBarPage();
```
Comment (adinapitul) [inline e2e/content-services/document-list/document-list-gallery.e2e.ts]: do we really need 2 NavigationBarPages, navBar and navigationBarPage ?

File: e2e/content-services/document-list/document-list-permissions.e2e.ts [hunk 25]
```diff
@@ -33,16 +33,16 @@ describe('Document List Component', () => {
     let privateSite;
     let acsUser = null;
 
-    beforeAll(() => {
+    beforeAll(async () => {
```
Comment (adinapitul) [inline e2e/content-services/document-list/document-list-permissions.e2e.ts]: async not needed

File: e2e/content-services/document-list/document-list-component.e2e.ts [hunk 26]
```diff
@@ -346,54 +345,54 @@ describe('Document List Component', () => {
...
             }
-            done();
+
         });
 
-        afterAll(async (done) => {
-            Promise.all(folderCreated.map((folder) =>
-                uploadActions.deleteFileOrFolder(folder.entry.id)
+        afterAll(async () => {
+            await Promise.all(folderCreated.map(async (folder) =>
```
Comment (adinapitul) [inline e2e/content-services/document-list/document-list-component.e2e.ts]: for..of might be safer to use ?

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts (hunk 4 above)
Comment (eromano) [reply lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: nope async needs to wrap te return in a promise this return already return a promise

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts (hunk 5 above)
Comment (eromano) [reply lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: nope async needs to wrap te return in a promise this return already return a promise

File: lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts (hunk 6 above)
Comment (eromano) [reply lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts]: see above

File: lib/testing/src/lib/core/pages/form/widgets/displayTextWidget.ts (hunk 7 above)
Comment (eromano) [reply lib/testing/src/lib/core/pages/form/widgets/displayTextWidget.ts]: see above

File: lib/testing/src/lib/process-services/pages/form-fields.page.ts (hunk 9 above)
Comment (eromano) [reply lib/testing/src/lib/process-services/pages/form-fields.page.ts]: not sure I got

File: lib/testing/src/lib/core/pages/form/formFields.ts (hunk 10 above)
Comment (eromano) [reply lib/testing/src/lib/core/pages/form/formFields.ts]: not sure I got is a promise

File: e2e/pages/adf/process-services/tasksListPage.ts [hunk 27]
```diff
@@ -17,48 +17,47 @@
...
+    taskList: ElementFinder = element(by.css('adf-tasklist'));
+    noTasksFound: ElementFinder = element.all(by.css("div[class='adf-empty-content__title']")).first();
+    dataTable: DataTableComponentPage = new DataTableComponentPage(this.taskList);
 
     getDataTable() {
         return this.dataTable;
     }
 
-    getRowsDisplayedWithSameName(taskName) {
+    getRowsDisplayedWithSameName(taskName): Promise<string> {
```
[the remark this answers is not in the document]
Comment (eromano) [reply e2e/pages/adf/process-services/tasksListPage.ts]: see above

File: e2e/pages/adf/contentServicesPage.ts (hunk 18 above)
Comment (eromano) [reply e2e/pages/adf/contentServicesPage.ts]: from the blame has been committed by you , I usually not comment code ;)

File: lib/testing/src/lib/core/pages/form/formFields.ts (hunk 10 above)
Comment (adinapitul) [reply lib/testing/src/lib/core/pages/form/formFields.ts]: sorry, should have escaped some chars, wanted to say that return type should be Promise of void, not string, according to the method name

File: e2e/pages/adf/contentServicesPage.ts (hunk 18 above)
Comment (adinapitul) [reply e2e/pages/adf/contentServicesPage.ts]: true, none the less, should be removed :D

File: e2e/core/login/login-sso/login-sso.e2e.ts [hunk 28]
```diff
@@ -18,6 +18,7 @@
 import { LoginSSOPage, SettingsPage, LoginPage } from '@alfresco/adf-testing';
 import { browser } from 'protractor';
 import { NavigationBarPage } from '../../../pages/adf/navigationBarPage';
+import { BrowserVisibility } from '../../../../lib/testing/src/lib/core/utils/browser-visibility';
```
Comment (adinapitul) [inline e2e/core/login/login-sso/login-sso.e2e.ts]: should be '@alfresco/adf-testing'

**Comment by eromano** [comment]
@adinapitul I have fixed all the comments except the one about async. see my comment. Some of your comment that I have implemented I guess where anyway not in the scope of this PR but they were more a name refactoring. I am ok with it but for now, I would like to postpone further improvements in a separate PR

**Comment by adinapitul** [comment]
regarding those asyncs... Take the following for example:
`    checkContentIsNotDisplayedByName(taskName): Promise<void> {
        return this.dataTable.checkContentIsNotDisplayed(column.name, taskName);
    }`
at this moment async indeed is not needed since there's no await inside.
which got me thinking as to WHY there's no await inside.
I think this method should look like this:
`async checkContentIsNotDisplayedByName(taskName): Promise<void> {
        await this.dataTable.checkContentIsNotDisplayed(column.name, taskName);
    }`
cause it doesn't really make sense to have return there.

**Comment by adinapitul** [comment]
to follow on my previous comment, what's the difference between
this: https://github.com/Alfresco/alfresco-ng2-components/blob/f4a0a4bc9320598418f697e54e3ae4e50d428ff9/lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts#L78
and this: https://github.com/Alfresco/alfresco-ng2-components/blob/f4a0a4bc9320598418f697e54e3ae4e50d428ff9/lib/testing/src/lib/content-services/pages/document-list.page.ts#L57 
?
we need to be consistent

**Pull request merged**